ext/drsyscall/drsyscall_record_trimmer.c

```diff
@@ -0,0 +1,186 @@
+    case DRSYS_PRECALL_PARAM:
+    case DRSYS_POSTCALL_PARAM:
+        return write_record((char *)record, sizeof(*record),
+                            (record->type == DRSYS_PRECALL_PARAM
+                                 ? "DRSYS_PRECALL_PARAM"
+                                 : "DRSYS_POSTCALL_PARAM"),
+                            current_record_timestamp);
+    case DRSYS_MEMORY_CONTENT:
+        if (!write_record((char *)record, sizeof(*record), "DRSYS_MEMORY_CONTENT record",
+                          current_record_timestamp)) {
+            return false;
+        }
+        return write_record(buffer, size, "DRSYS_MEMORY_CONTENT content",
+                            current_record_timestamp);
+    case DRSYS_RETURN_VALUE:
+        return write_record((char *)record, sizeof(*record), "DRSYS_RETURN_VALUE",
+                            current_record_timestamp);
+    case DRSYS_RECORD_END_TIMESTAMP:
+        if (!write_record((char *)record, sizeof(*record), "DRSYS_RECORD_END_TIMESTAMP",
+                          current_record_timestamp)) {
+            return false;
+        }
+        current_record_timestamp = record->syscall_number_timestamp.timestamp;
+        return true;
+    default: dr_printf("unknown record type %d\n", record->type); return false;
+    }
+    return true;
+}
+
+int
+main(int argc, char *argv[])
+{
+    if (argc < 2) {
+        usage(argv[0]);
+        return -1;
+    }
+
+    int arg_index = 1;
+    while (arg_index < argc) {
+        if (!strncmp(argv[arg_index], "--help", 6)) {
+            usage(argv[0]);
+            return 0;
+        }
+        if (!strncmp(argv[arg_index], "-i", 2)) {
+            record_file = open(argv[++arg_index], O_RDONLY);
+            if (record_file == -1) {
+                dr_printf("unable to open input file %s.\n", argv[arg_index]);
+                return -1;
+            }
+
+        } else if (!strncmp(argv[arg_index], "-o", 2)) {
+            output_file =
+                open(argv[++arg_index], O_CREAT | O_TRUNC | O_WRONLY, S_IRUSR | S_IWUSR);
+            if (output_file == -1) {
+                dr_printf("unable to open output file %s.\n", "blah");
+                return -1;
+            }
+        } else if (!strncmp(argv[arg_index], "-a", 2)) {
+            char *end_ptr;
+            trim_after_timestamp = strtoull(argv[++arg_index], &end_ptr, /*base=*/10);
+            if (end_ptr == NULL) {
+                dr_printf("invalid trim after timestamp: %s.\n", argv[arg_index]);
+                return -1;
+            }
+        } else if (!strncmp(argv[arg_index], "-b", 2)) {
+            char *end_ptr;
+            trim_before_timestamp = strtoull(argv[++arg_index], &end_ptr, /*base=*/10);
+            if (end_ptr == NULL) {
+                dr_printf("invalid trim before timestamp: %s.\n", argv[arg_index]);
+                return -1;
+            }
+        }
+        ++arg_index;
+    }
+
+    if (record_file == -1) {
```

Comment:
Is the record file a simple binary file?

 Not for this PR: just curious if we plan to or need to use the zip format which can allow fast-skipping.